pkg/services/sqlstore/datasource_test.go

```diff
@@ -3,173 +3,212 @@ package sqlstore
 import (
 	"testing"
 
-	. "github.com/smartystreets/goconvey/convey"
-
 	"github.com/grafana/grafana/pkg/models"
+	"github.com/stretchr/testify/assert"
 )
 
-type Test struct {
-	Id   int64
-	Name string
-}
-
 func TestDataAccess(t *testing.T) {
-	Convey("Testing DB", t, func() {
-		InitTestDB(t)
-		Convey("Can add datasource", func() {
+	defaultAddDatasourceCommad := models.AddDataSourceCommand{
+		OrgId:  10,
+		Name:   "nisse",
+		Type:   models.DS_GRAPHITE,
+		Access: models.DS_ACCESS_DIRECT,
+		Url:    "http://test",
+	}
+
+	defaultUpdateDatasourceCommad := models.UpdateDataSourceCommand{
+		OrgId:  10,
+		Name:   "nisse_updated",
+		Type:   models.DS_GRAPHITE,
+		Access: models.DS_ACCESS_DIRECT,
+		Url:    "http://test",
+	}
+
+	initDatasource := func() *models.DataSource {
+		cmd := defaultAddDatasourceCommad
+		err := AddDataSource(&cmd)
+		assert.Nil(t, err)
```

Comment:
You should use `require` instead of `assert` throughout the test, until the very end, because `assert` will continue the test after an error, which will lead to chaos:
```suggestion
		require.Nil(t, err)
```

Reply:
What?! I did not know that. I am used to assertions being ... well assertions so something that when fails stops the execution. I assume more people does not know about that cause I think we have assert everywhere in the go tests.